docs/docs/behaviors/dynamic-macros.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Dynamic Macro Behavior
+sidebar_label: Dynamic Macros
+---
+
+## Summary
+
+The dynamic macro behavior allows creating macros and replaying them by recording key presses. While recording a macro, you can also play another macro.
+
+:::note
+Dynamic macros are cleared on reboot.
+:::
+
+:::warning
+Dynamic macros are memory intensive, and may cause the firmware to crash. It is recommended to use only 1 dynamic macro in your keymap and re-record it if it needs changed. The maximum amount of actions to be recorded can be set with `CONFIG_ZMK_DYNAMIC_MACRO_MAX_ACTIONS` (default 64).
+:::
+
+## Dynamic Macro Action Defines
+
+Dynamic macro action defines are provided through the [`dt-bindings/zmk/dynamic-macros.h`](https://github.com/zmkfirmware/zmk/blob/main/app/include/dt-bindings/zmk/dynamic-macros.h) header,
+which is added at the top of the keymap file:
+
+```
+#include <dt-bindings/zmk/dynamic-macros.h>
+```
+
+This will allow you to reference the actions defined in this header such as `PLAY`.
+
+Here is a table describing the action for each define:
+
+| Define   | Action                      |
+| -------- | --------------------------- |
+| `PLAY`   | Play back a recorded macro  |
+| `RECORD` | Toggle recording of a macro |
```

Comment:
How does it know when to stop recording?

Reply:
When you press RECORD again

Reply:
I've been trying this for a while, and I believe it only works the first time.  I can record, stop the record, and play it numerous times.  But when I try to record again, it hangs, and I'll need to unplug the keyboard again.

Reply:
I can not replicate this. I have been using for months on the BT60v1 and there is no problem with multiple uses. I am building with the latest 3.2 upgrade now, but I was also using it before without any trouble.